### DISCUSSION TITLE
Filament and Octane

### DISCUSSION THREAD

**Discussion by fpolli** [filamentphp/filament#14210] [General]
I have seen a few issues and discussions involving Octane that indicate to me that Filament has some architectural inconsistencies with Octane, that some people are using the two packages together, and that the creator of Filament is working hard to make it work with Octane.

I have noticed in the source that Filament relies a lot on static properties, particularly in resource files, and I have read frequently that static properties should be avoided with Octane. I am wondering how these are reconciled. Do I need to worry about the use of static properties in Filament? What measures will I need to take to make sure that my implementation of Filament works correctly with Octane?

Thank you.

**Comment by indigoram89** [comment]
+1

**Comment by fpolli** [comment]
Does anybody who has insight into the structure of the code have any guidance on using Filament with Octane? What gets loaded and instantiated during application bootup and what gets done during request runtime? From early attempts to do things, it appears that the panel service providers are set during application bootup and cannot be customized on a per-request basis. I'm trying to set a lot of values based on the request, and nothing put in the panel() function of a panel service provider reflects those changes. It appears to be pulling values as they are at app bootup and that is fatal to my project. It appears I may have just wasted two weeks and a lot of code on a framework I cannot actually use.
